### ISSUE TITLE
Can I determine the optimal combination of parameters at first, then increase the number of training data?

### ISSUE THREAD

**Issue by take0212** [optuna/optuna#2772]
Can I determine the optimal combination of parameters at first, then increase the number of training data?

Procedure
1. Determine the optimal combination of parameters  by using optuna with a half of train data and valid accuracy.
2. Retrain with the optimal combination of parameters and all train data.

In this case, when I use all train data, the valid loss may be minimized at smaller epochs. I can use early stopping as a countermeasure, but I'm not sure whether it is the best countermeasure.

Questions
1. When I increase the number of training data, is it correct to decrease the learning rate?
2. When I use accuracy as an indicator, is it sufficient to use only valid accuracy, or should I use both of  valid accuracy and train accuracy?

If you have any advice, please let me know.
Thank you very much.

**Comment by nzw0301** [comment]
I suppose that this issue asking about general questions related to hyper-parameter optimisation for machine learning rather than optuna specific questions. Anyway, I'll answer the questions.

## 1. Can I determine the optimal combination of parameters at first, then increase the number of training data?

Yes. Indeed, the same technique is implemented by a popular machine learning library such as scikit-learn. Precisely, `refit` argument in [`sklearn.model_selection.GridSearchCV`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html) enables us to train the model on the whole dataset with hyper-parameters by using cross-validation.
But it might not be optimal anymore since the training dataset at the second step is not the same as in the first step.
See also Sec 7.8 in [Deep Learning](https://www.deeplearningbook.org/) book for more detailed discussion.

## 2. When I increase the number of training data, is it correct to decrease the learning rate?

I think, no. Could you explain the reason why the learning rate should decrease 

In my understanding, the learning rate is also a hyperparameter. If only the number of training samples increases, the learning rate should not be changed.
Otherwise, I suppose that the optimisation result changes drastically.

## 3. When I use accuracy as an indicator, is it sufficient to use only valid accuracy, or should I use both of valid accuracy and train accuracy?

Depending on the application, but probably I think the validation accuracy is enough. I'm not sure how to use the training accuracy for hyperparameter optimisation...

**Comment by take0212** [comment]
Thank you for your comment.

Sorry for asking a general question about optimization.
I expect that optuna users have similar experience or knowledge.

1.About sklearn.model_selection.GridSearchCV
My implementation is different, so I can't use it right away.
Thanks for your advice.

2.About lowering the initial value of the learning rate
When I increase the number of training data, the valid loss is minimized in fewer epochs. Early stopping can be used as a countermeasure, but iteration is not increased. So it may be not good way.
When I lower the initial value of the learning rate, early stopping may not happen.  I may increase the number of iterations by increasing the number of training data.  So it may be good way.

3.About using train accuracy together
When I check the train accuracy of the top optimal results with valid accuracy, I can see that it varies.
I think that a model with the high train accuracy would be more suitable when the number of training data is increased.

4. About the ideal search method
Because Optuna is "defined by run", it is better to add the number of training data to the hyperparameters, increase the search range of the number of training data gradually and make the hyperparameters shift to the best hyperparameters for the max value of the training data gradually.
The procedure will reduce the search time compared to searching with all training data from the first search.
It would be nice if the hyperparameters selected for a small number of training data could be converted to the best hyperparameters for all training data without much effort.

I 'm training image data.

**Comment by nzw0301** [comment]
Thank you for your explanation. In terms of question 2, I think that the training dataset at the second step is largely different from the training dataset at the first step, so the suggested hyperparameters probably are not optimal in practice. 

I do not recommend treating the number of training samples as a hyper-parameter because I suppose that the obtained hyper-parameters perform well among the different training datasets whose size is different, not the full size of the dataset.

Best regards,
